store/s3/s3.go

```diff
@@ -39,11 +41,11 @@ func New(uri string) (*Store, error) {
 	u, err := url.Parse(uri)
 
 	if err != nil {
-		return nil, fmt.Errorf("%w", err)
+		return nil, wrapError(fmt.Sprintf("failed to parse URI %s", uri), err)
 	}
 
 	if u.Scheme != "s3" {
-		return nil, fmt.Errorf("Invalid uri scheme for s3")
+		return nil, wrapError(fmt.Sprintf("invalid S3 scheme in URI %s. Scheme is %s", uri, u.Scheme), fmt.Errorf("Invalid S3 URI"))
```

Comment:
yeah... this wrapError doesn't make any sense.
fmt.Errorf could do all in a single pass and more efficiently.